src/clib/pio.h

```diff
@@ -1193,6 +1193,9 @@ typedef struct file_desc_t
     /** True if we need reserve some extra space in the header when
      * creating NetCDF files to accommodate anticipated changes. */
     bool reserve_extra_header_space;
+
+    /** True if this file is opened to read or write */
+    bool is_opened;
```

Comment:
Are there scenarios where append is done without opening a file?

Reply:
No, I haven’t encountered any cases where append is performed without first opening the file. In fact, if a file is not opened before an append operation, our existing code fails to retrieve a valid file ID and returns an error earlier. So, in practice, appending without opening is not a valid scenario.

Reply:
So do we need this extra bool or can we glean the info from the id or some other state?

Reply:
Currently, we don’t have any other flags in file_desc_t that indicate whether a file was newly created or opened. This is_opened flag fills that gap and will also be useful for future development where this distinction may matter.

Reply:
Still not sure why we need this new variable (bio_get_file(), file id etc give the same information, right? Also these sanity checks are done before we enter the code that appends data, so why do we need this extra var?)

Reply:
I’ll go with is_reopened and update the commit messages accordingly.